### PR TITLE
chore: convert to TS

### PR DESCRIPTION
**Pull request by SimenB**
Take 2 of the TS migration.

~Needs to wait for https://github.com/typescript-eslint/typescript-eslint/pull/712 to avoid peer dep warning when installing this, but I _think_ this now works beyond that 🙂~

~^ is released as canary. We should either wait for a stable or their next major, then we can merge this 🤞 ~

^ Out as 1.13, will merge this

/cc @jeysal

### CODE REVIEW

File: src/rules/__tests__/lowercase-name.test.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
-import { RuleTester } from 'eslint';
+import { TSESLint } from '@typescript-eslint/experimental-utils';
 import rule from '../lowercase-name';
 
-const ruleTester = new RuleTester({
+const ruleTester = new TSESLint.RuleTester({
+  parser: '@typescript-eslint/parser',
```
Comment (SimenB) [inline src/rules/__tests__/lowercase-name.test.ts]: I'd still like to get rid of this

**Comment by SimenB** [comment]
@G-Rath I'll rebase this after merging #255 - you might need to rebase your PRs as well

**Comment by G-Rath** [comment]
@SimenB you're welcome to rebase them - I suspect you're better at it anyway 😉

**Comment by SimenB** [comment]
Cool, I'll rebase all of your open PRs, then 🙂 I'll ping you if tests fail in any of them and they're not trivial to fix

**Pull request merged**